### PR TITLE
Resolving #25 through spacing overhaul

### LINKED ISSUE #25: Remove spacing form components
Please remove spacing from the outside of components, this will allow easier positioning in layouts for teams.

### PR DESCRIPTION
**Pull request by jmillar-nanthealth**
- [x] Converts to REM
- [x] Removes default spacing from components
- [x] Adds utility classes for various spacing types similar to other libs

### CODE REVIEW

File: docs/.vuepress/global-components/FeatherDemo.vue [hunk 1]
```diff
@@ -122,11 +122,13 @@ export default {
 @import "@featherds/styles/themes/variables";
 @import "@featherds/styles/mixins/typography";
 @import "@featherds/styles/mixins/responsive";
+@import "@featherds/input-helper/scss/spacing";
 .feather-demo {
+  //@include form-spacing();
```
Comment (rikbrowning) [inline docs/.vuepress/global-components/FeatherDemo.vue]: can this be removed and the import?

File: docs/Components/DateInput/README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ A Date Input consists of two primary methods of interaction:
 
 The Date Input is a “spinner” control that appears as a text field and allows for a granular level of keyboard control with the arrow keys. The text field Date Input is best used when the user needs to enter a date from the distant past or far in the future. The text field date input allows for faster input thanks to the minimal number of clicks involved and the speed with which users can type.
 
-The Date picker is a graphical calendar that involves more clicks on the user’s behalf but provides a spatial context for the individual day and its relationship to all other days in each month. This can provide more context for a user but requires more cognitive overhead when used
+The Date Input is a graphical calendar that involves more clicks on the user’s behalf but provides a spatial context for the individual day and its relationship to all other days in each month. This can provide more context for a user but requires more cognitive overhead when used
```
Comment (rikbrowning) [inline docs/Components/DateInput/README.md]: The picker is the calendar? surely the date input is the spin buttons?

File: packages/@featherds/styles/themes/_base-light.scss [hunk 3]
```diff
@@ -14,19 +14,57 @@ $colors: (
...
+  $hero-gradient-2-4:
+    radial-gradient(
+      ellipse 200% 200% at -90% 160%,
+      rgba(#3d50c5, 0.5) 0%,
+      #14d1df00 100%
+    )
+    0% 0% no-repeat padding-box,
 );
 
 $on-color: (
```
Comment (rikbrowning) [inline packages/@featherds/styles/themes/_base-light.scss:70]: don't think we should change these values to rem

File: packages/@featherds/megamenu/src/components/FeatherMegaMenu.vue [hunk 4]
```diff
@@ -155,14 +155,14 @@ export default {
...
-    margin-bottom: 24px;
+    margin-top: 2rem;
+    margin-left: 2rem;
+    margin-right: 0.75rem;
+    margin-bottom: 1.5rem;
     width: 100%;
     @include fill;
     & :deep(li > a) {
-      border-radius: 4px;
+      border-radius: 0.25rem;
```
Comment (rikbrowning) [inline packages/@featherds/megamenu/src/components/FeatherMegaMenu.vue]: Not sure we should change border radius

File: packages/@featherds/input-helper/scss/_spacing.scss [hunk 5]
```diff
@@ -1,31 +1,84 @@
 @import "@featherds/styles/themes/variables";
```
Comment (rikbrowning) [inline packages/@featherds/input-helper/scss/_spacing.scss:1]: can this file not go in styles? can apply to anything not just input?

File: docs/Components/DateInput/README.md (hunk 2 above)
Comment (jmillar-nanthealth) [reply docs/Components/DateInput/README.md]: Good point. I will change this to "date picker"

File: packages/@featherds/megamenu/src/components/FeatherMegaMenu.vue (hunk 4 above)
Comment (jmillar-nanthealth) [reply packages/@featherds/megamenu/src/components/FeatherMegaMenu.vue]: Border radii all reverted to px values

File: docs/.vuepress/global-components/FeatherDemo.vue (hunk 1 above)
Comment (jmillar-nanthealth) [reply docs/.vuepress/global-components/FeatherDemo.vue]: Thanks!

File: packages/@featherds/styles/themes/_base-light.scss (hunk 3 above)
Comment (jmillar-nanthealth) [reply packages/@featherds/styles/themes/_base-light.scss:70]: Resolved

File: packages/@featherds/input-helper/scss/_spacing.scss (hunk 5 above)
Comment (jmillar-nanthealth) [reply packages/@featherds/input-helper/scss/_spacing.scss:1]: Moved to the general styles

**Pull request merged**